domain/crossmodelrelation/state/model/remoteapplication.go

```diff
@@ -1540,3 +1568,61 @@ AND    e2.endpoint_name    = $endpointIdentifier2.endpoint_name
 
 	return relationsSuspended[0].IsSuspended, errors.Capture(err)
 }
+
+func (st *State) isRelationCrossModel(ctx context.Context, tx *sqlair.TX, key corerelation.Key) (bool, error) {
+	// Get the relation UUID from the key (2-endpoint relation).
+	// The key contains endpoint identifiers (application names and endpoint names).
+	endpoints := key.EndpointIdentifiers()
+
+	// Use the existing method to get the relation UUID.
+	// This reuses logic already present in relation_network.go.
+	relationUUID, err := st.getRegularRelationUUIDByEndpointIdentifiers(ctx, tx, endpoints[0], endpoints[1])
+	if err != nil {
+		if errors.Is(err, relationerrors.RelationNotFound) {
+			return false, errors.Errorf("relation not found for key %q: %w", key.String(), relationerrors.RelationNotFound)
+		}
+		return false, errors.Errorf("getting relation UUID: %w", err)
+	}
+
+	// Check if any application in this relation is remote.
+	isRemote, err := st.isApplicationRemote(ctx, tx, relationUUID)
+	if err != nil {
+		return false, errors.Errorf("checking if application is remote: %w", err)
+	}
+
+	return isRemote, nil
+}
+
+func (st *State) isApplicationRemote(ctx context.Context, tx *sqlair.TX, relationUUID string) (bool, error) {
```

Comment:
nit: is _application_ remote but takes in a relation uuid?

Reply:
```suggestion
func (st *State) relationContainsRemoteApplication(ctx context.Context, tx *sqlair.TX, relationUUID string) (bool, error) {
```

I think that'd be a better name.